Assets/Scripts/UIAndMenus/EscMenuController.cs

```diff
@@ -13,26 +18,37 @@ void Start()
     {
         pauseMenuPanel.SetActive(false);
         settingsMenuPanel.SetActive(false);
+        DontDestroyOnLoad(gameObject);
+    }
+
+    private void OnEnable()
+    {
+        pauseAction.performed += PauseToggle;
+        pauseAction.Enable();
     }
 
-    public void MenuToggle()
+    private void OnDisable()
     {
-        if(pauseMenuPanel.activeSelf)
+        pauseAction.performed -= PauseToggle;
+        pauseAction.Disable();
+    }
+
+    public void PauseToggle(CallbackContext c)
+    {
+        if(GameStateManager.Instance.CurrentState is PlayingState)
         {
-            if (settingsMenuPanel.activeSelf)
-            {
-                CloseSettings();
-            }
-            pauseMenuPanel.SetActive(false);
-            Time.timeScale = 1f;
-        }
-        else
+            GameStateManager.Instance.SetState(new GameStateManagement.PauseState(this));
+        } else
         {
-            pauseMenuPanel.SetActive(true);
-            Time.timeScale = 0f;
+            GameStateManager.Instance.SetState(new PlayingState());
         }
     }
```

Comment:
Should add a check to determine whether in the main menu (currently, you can access the pause screen when in the main menu, which should not be possible)